compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/lower/JvmInlineClassLowering.kt

```diff
@@ -433,7 +444,9 @@ private class JvmInlineClassLowering(context: JvmBackendContext) : JvmValueClass
             copyParameterDeclarationsFrom(irConstructor)
             annotations = irConstructor.annotations
             body = context.createIrBuilder(this.symbol).irBlockBody(this) {
-                +irDelegatingConstructorCall(context.irBuiltIns.anyClass.owner.constructors.single())
+                +irDelegatingConstructorCall(valueClass.superTypes.single {
+                    it.classOrNull?.owner?.kind == ClassKind.CLASS
+                }.classOrNull!!.constructors.single().owner)
```

Comment:
Yeah. Sorry. The problem here is that design has changed middle-way and first commits became obsolete. To avoid the confusion, I should have squashed the commits. I did it now and split it into several relevant parts.